src/Controls/src/Core/Window/Window.cs

```diff
@@ -431,7 +431,13 @@ void SendWindowAppearing()
 		void SendWindowDisppearing()
 		{
 			if (Navigation.ModalStack.Count == 0)
+			{
 				Page?.SendDisappearing();
+			}
+			else if (Navigation.ModalStack.Count > 0)
+			{
+				Navigation.ModalStack[Navigation.ModalStack.Count - 1]?.SendDisappearing();
```

Comment:
Can use `Navigation.ModalStack[^1];`

Reply:
@jsuarezruiz I’ve updated the code based on your suggestion. Please let me know if you have any further concerns.